nntrainer/src/neuralnet.cpp

```diff
@@ -346,13 +348,69 @@ int NeuralNetwork::init() {
       conv2d_layer->setWeightDecay(weight_decay);
       NN_INI_RETURN_STATUS();
 
-      status = conv2d_layer->setOptimizer(opt);
+      status = conv2d_layer->initialize(last);
       NN_INI_RETURN_STATUS();
 
-      status = conv2d_layer->initialize(last);
+      status = conv2d_layer->setOptimizer(opt);
       NN_INI_RETURN_STATUS();
+
       layers.push_back(conv2d_layer);
     } break;
+
+    case LAYER_POOLING2D: {
+      int size[POOLING2D_DIM];
+      std::shared_ptr<Pooling2DLayer> pooling2d_layer =
+        std::make_shared<Pooling2DLayer>();
+
+      pooling2d_layer->setInputDimension(previous_dim);
+
+      status =
+        getValues(POOLING2D_DIM,
+                  iniparser_getstring(
+                    ini, (layer_name + ":pooling_size").c_str(), unknown),
+                  (int *)size);
+
+      NN_INI_RETURN_STATUS();
+      status =
+        pooling2d_layer->setSize(size, Layer::PropertyType::pooling_size);
+
+      NN_INI_RETURN_STATUS();
+      status = getValues(
+        POOLING2D_DIM,
+        iniparser_getstring(ini, (layer_name + ":stride").c_str(), unknown),
```

Comment:
lets not use unknown, but sensible defaults - getValues({1, 1}).

Reply:
fixed. thanks.